### PR TITLE
New resource: dedicated_host_group (compute)

### PR DESCRIPTION
**Pull request by magodo**
Add new resource `dedicated_host_group`, which belongs to Compute Provider.

> A host group is a resource that represents a collection of dedicated hosts. You create a host group in a region and an availability zone, and add hosts to it.

Please refer to this [link](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/dedicated-hosts#groups-hosts-and-vms) for more info.

### CODE REVIEW

File: website/docs/d/dedicated_host_group.html.markdown [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (katbyte) [inline website/docs/d/dedicated_host_group.html.markdown]: Could we add this to the index.erb

File: website/docs/r/dedicated_host_group.html.markdown [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
```
Comment (katbyte) [inline website/docs/r/dedicated_host_group.html.markdown]: Could we add this file to the index.erb

**Comment by magodo** [comment]
@katbyte, this PR is filed before the refactor you've made. Now I think I'd better to reflect the refactoring to this resource with your comment applied and raise another PR.

File: azurerm/data_source_dedicated_host_group.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroupName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if dedicatedHostGroupProperties := resp.DedicatedHostGroupProperties; dedicatedHostGroupProperties != nil {
+		if err := d.Set("hosts", flattenArmDedicatedHostGroupSubResourceReadOnly(dedicatedHostGroupProperties.Hosts)); err != nil {
+			return fmt.Errorf("Error setting `hosts`: %+v", err)
+		}
+		d.Set("platform_fault_domain_count", int(*dedicatedHostGroupProperties.PlatformFaultDomainCount))
```
[the remark this answers is not in the document]
Comment (magodo) [reply azurerm/data_source_dedicated_host_group.go]: Why is that? I assume a `schema.TypeInt` corresponds to `int`, not `int32`.

**Comment by magodo** [comment]
Replaced by #5307

**Pull request closed without merging**